### PR TITLE
fix: option key does not work in addClickShortCut() method

### PR DESCRIPTION
**Pull request by czp13**
## Description

This is a draft PR so do not merge please until we did not discuss it thoroughly.

Contex:
In Firefox, the option key is not working. The reason behind this as the browser implementing differently the KeyboardEvent when the option key is pushed, and when getModifierState() is called it will give back different true or false events based on whether you are using Chrome or Firefox.

The difference comes down to this:

Mozzila:
```
event.getModifierState('Alt') -> true
event.getModifierState('AltGraph') -> true
```

Chrome:
```
event.getModifierState('Alt') -> true
event.getModifierState('AltGraph') -> false
```

For more information:
Check to please the table for `When getModifierState() returns true on Gecko?` here:
- https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/getModifierState

Fixes # (issue)
https://github.com/vaadin/flow/issues/7798

## Type of change

- [x] Bugfix
- [ ] Feature

## Checklist

- [x] I have read the contribution guide: https://vaadin.com/docs/latest/guide/contributing/overview/
- [x] I have added a description following the guideline.
- [x] The issue is created in the corresponding repository and I have referenced it.
- [x] I have added tests to ensure my change is effective and works as intended.
- [x] New and existing tests are passing locally with my change.
- [x] I have performed self-review and corrected misspellings.

#### Additional for `Feature` type of change

- [ ] Enhancement / new feature was discussed in a corresponding GitHub issue and Acceptance Criteria were created.

### CODE REVIEW

File: flow-server/src/main/java/com/vaadin/flow/component/ShortcutRegistration.java [hunk 1]
```diff
@@ -50,18 +50,18 @@ public class ShortcutRegistration implements Registration, Serializable {
...
+                    + "delegate.addEventListener('keydown', function(event) {"
+                    + "if (%2$s) {" // the filter text to match the key
+                    + "const new_event = new event.constructor(event.type, event);"
+                    + "listenOn.dispatchEvent(new_event);"
+                    + "%3$s" // the new event allows default if desired
+                    + "event.stopPropagation();}" // the new event bubbles if desired
+                    + "});" // end matches filter
+                    + "} else {"
+                    + "throw \"Shortcut listenOn element not found with JS locator string '%1$s'\""
+                    + "}";//@formatter:on
```
Comment (tepi) [inline flow-server/src/main/java/com/vaadin/flow/component/ShortcutRegistration.java:64]: Only indentation change?

Comment (czp13) [reply flow-server/src/main/java/com/vaadin/flow/component/ShortcutRegistration.java:64]: Thanks, @tepi! Yes, only indentation. What do you feel, shall I change back?
I usually push reformat on files I touch.

Comment (tepi) [reply flow-server/src/main/java/com/vaadin/flow/component/ShortcutRegistration.java:64]: Well I think it's fine as long as it's from `mvn formatter:format`

Comment (czp13) [reply flow-server/src/main/java/com/vaadin/flow/component/ShortcutRegistration.java:64]: not sure if it was my format action :D or `mvn formatter:format.`, but I executed the mvn command one as well many times, and if this change goes through then probably it is aligned with format rules.

ok, we can let it stay :)

**Review by tepi — DISMISSED** [review]

**Review by tepi — APPROVED** [review]

**Pull request merged**